### PR TITLE
Streamhandler create response

### PR DESCRIPTION
**Pull request by anwar6953**
Stream handler test cases were not being executed. This caused an API change (response code) to go through to develop, without test case failure (https://github.com/caskdata/cdap/pull/2752).

This PR resolves (reverts) that response code change.
It also merges StreamHandlerTestV3, since we no longer have a V2 and V3 version of those.

http://builds.cask.co/browse/CDAP-DUT2001-2

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java [hunk 1]
```diff
@@ -188,8 +187,14 @@ public void create(HttpRequest request, HttpResponder responder,
...
     namespaceClient.get(namespaceId);
 
-    // Verify stream name
-    Id.Stream streamId = Id.Stream.from(namespaceId, stream);
+    Id.Stream streamId;
+    try {
+      // Verify stream name
+      streamId = Id.Stream.from(namespaceId, stream);
+    } catch (IllegalArgumentException e) {
+      LOG.error("Failed to create stream {}", stream, e);
```
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java]: Don't log error (in fact, no need to log). If really want to log, log as debug or trace. A user error shouldn't result in ERROR level log.

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/StreamHandlerTest.java [hunk 2]
```diff
@@ -253,4 +277,142 @@ public void testPutInvalidStreamConfig() throws Exception {
...
+    Id.Stream streamId1 = Id.Stream.from(TEST_NAMESPACE1, streamName);
+    Id.Stream streamId2 = Id.Stream.from(TEST_NAMESPACE2, streamName);
+
+    createStream(streamId1);
+    createStream(streamId2);
+
+    List<String> eventsSentToStream1 = Lists.newArrayList();
+    // Enqueue 10 entries to the stream in the first namespace
+    for (int i = 0; i < 10; ++i) {
+      String body = streamId1.getNamespaceId() + Integer.toString(i);
```
Comment (chtyim) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/StreamHandlerTest.java]: Just write `streamId1.getNamespaceId() + i`

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/StreamHandlerTest.java [hunk 3]
```diff
@@ -253,4 +277,142 @@ public void testPutInvalidStreamConfig() throws Exception {
...
+      sendEvent(streamId1, body);
+      eventsSentToStream1.add(body);
+    }
+
+    List<String> eventsSentToStream2 = Lists.newArrayList();
+    // Enqueue only 5 entries to the stream in the second namespace, decrementing the value each time
+    for (int i = 0; i > -5; --i) {
+      String body = streamId1.getNamespaceId() + Integer.toString(i);
```
Comment (chtyim) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/StreamHandlerTest.java]: Same here, just `+ i`.

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/StreamHandlerTest.java [hunk 4]
```diff
@@ -253,4 +277,142 @@ public void testPutInvalidStreamConfig() throws Exception {
...
+    return events;
+  }
+
+  private void checkEventsProcessed(Id.Stream streamId, long expectedCount, int retries) throws Exception {
+    for (int i = 0; i < retries; i++) {
+      long numProcessed = getNumProcessed(streamId);
+      if (numProcessed == expectedCount) {
+        return;
+      }
+      TimeUnit.SECONDS.sleep(1);
```
Comment (chtyim) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/StreamHandlerTest.java]: Just `Tasks.waitFor` with sleep time of 100ms to have faster test.

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/StreamHandlerTest.java [hunk 5]
```diff
@@ -253,4 +277,142 @@ public void testPutInvalidStreamConfig() throws Exception {
...
+    HttpResponse response = HttpRequests.execute(request);
+    Assert.assertEquals(200, response.getResponseCode());
+  }
+
+  private List<String> fetchEvents(Id.Stream streamId) throws Exception {
+    URL url = createURL(streamId.getNamespaceId(), "streams/" + streamId.getId() + "/events");
+    HttpRequest request = HttpRequest.get(url).build();
+    HttpResponse response = HttpRequests.execute(request);
+    Assert.assertEquals(200, response.getResponseCode());
+
```
Comment (chtyim) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/StreamHandlerTest.java]: Remove extra empty line. Just one is enough.

**Comment by chtyim** [comment]
Couple minor comments, rest LGTM.

**Comment by anwar6953** [comment]
@chtyim Updated.

**Comment by chtyim** [comment]
LGTM

**Comment by anwar6953** [comment]
Pushed updates to fix two failing test cases (they were outdated, since they had not run in a while).

**Comment by chtyim** [comment]
LGTM. Merge when unit-tests passed.

**Pull request merged**